server/oauth_service.py

```diff
@@ -0,0 +1,45 @@
+import aiocron
+import jwt
+
+from jwt import PyJWKClient, InvalidTokenError
+
+from server.config import config
+from .core import Service
+from .decorators import with_logger
+from .exceptions import AuthenticationError
+
+
+@with_logger
+class OauthService(Service):
+    """
+        Service for managing the OAuth token logins and verification.
+    """
+
+    def __init__(self):
+        self.jwks_client = None
+
+    async def initialize(self) -> None:
+        await self.retrieve_public_keys()
+        # crontab: min hour day month day_of_week
+        # Run every day to update public keys.
+        self._update_cron = aiocron.crontab(
+            "0 0 * * *", func=self.retrieve_public_keys
+        )
+
+    async def retrieve_public_keys(self) -> None:
+        """
+            Get the latest jwks from the hydra endpoint
+        """
+        self.jwks_client = PyJWKClient(config.HYDRA_JWKS_URI)
```

Comment:
Good news and bad news:
Bad news - this uses urllib which is blocking and pyjwt does not have any features for async
Good news - this class is pretty simple so it shouldn't be a big deal to write our own async version of it, or maybe just have the OAuthService do that directly
https://github.com/jpadilla/pyjwt/blob/69d1e8b5f450f2b1df42149d977b48e286df1054/jwt/jwks_client.py

Reply:
Hmm okay then I will just use aiohttp then probably